### PR TITLE
vrb: Reduce storage pressure from req.body

### PR DESCRIPTION
**Pull request by Dridi**
Instead of trying to allocate storage for the entire request body when the content length is known, let it go in fetch_chunksize steps. This may prevent spurious cache evictions for large request bodies for pass transactions when we are merely streaming the body.

On the other hand, when the content length is known and the goal is to cache the request body, we may still attempt a single allocation and all that entails.

### CODE REVIEW

File: bin/varnishd/cache/cache_req_body.c
Comment (Dridi) [inline bin/varnishd/cache/cache_req_body.c:129]: Dead branch spotted by @nigoroll.

**Comment by Dridi** [comment]
bugwash: waiting for @nigoroll's approval.

I will first double check the dead branch, and CI failure for bullseye (job with sanitizers).

File: bin/varnishd/cache/cache_req_body.c
Comment (Dridi) [reply bin/varnishd/cache/cache_req_body.c:129]: `l` may be increased by `VFP_GetStorage()` and become larger than `yet`, in which case this branch may be reached when the remaining bytes fit entirely in the allocated buffer.

**Comment by Dridi** [comment]
The sanitizer failure also happens on the master branch, and could probably be solved with a revert of 20824dc0f75c71dbcd0a2edc83bc88c1b8293641 modulus a conflict resolution. But this isn't straightforward since this may break other CI jobs (we can always stick a `feature 64bit` with a comment).

Ready to merge with no reserves on my end after @nigoroll's approval.

File: bin/varnishd/cache/cache_req_body.c
Comment (nigoroll) [reply bin/varnishd/cache/cache_req_body.c:129]: Oh, sorry, I confused the `l` for a `1`.

**Review by nigoroll — APPROVED** [review]

**Pull request merged**
